cmd/tempo-query/tempo/plugin.go

```diff
@@ -306,6 +312,32 @@ func (b *Backend) GetOperations(ctx context.Context, _ *storage_v1.GetOperations
 	}, nil
 }
 
+type job struct {
+	ctx     context.Context
+	traceID jaeger.TraceID
+}
+
+type jobResult struct {
+	traceID jaeger.TraceID
+	trace   *jaeger.Trace
+	err     error
+}
+
+func worker(b *Backend, jobs <-chan job, results chan<- jobResult, wg *sync.WaitGroup) {
+	for job := range jobs {
+		jaegerTrace, err := b.getTrace(job.ctx, job.traceID)
+		if err != nil {
+			b.logger.Info("failed trace", zap.Error(err), zap.String("traceid", job.traceID.String()))
+		}
+		results <- jobResult{
+			traceID: job.traceID,
+			trace:   jaegerTrace,
+			err:     err,
+		}
+		wg.Done()
```

Comment:
See the note about removing `wg` from the passed arguments on this function.  Otherwise I think we want to defer the `wg.Done()` call at the top of this function.